File: 643-MaximumAverageSubarrayI/643-MaximumAverageSubarrayI.py
# Last updated: 13/07/2026, 22:17:47
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        left = 0
        right = k
        cur = sum(nums[:k])
        result = cur
        
        while (right < len(nums)):
            cur = cur - nums[left] + nums[right]
            result = max(cur, result)
            right += 1
            left += 1
        
        return result / k

File: 643-MaximumAverageSubarrayI/643-MaximumAverageSubarrayI.cpp
// Last updated: 13/07/2026, 22:17:41
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double total = 0;
        for (int i = 0; i < k; ++i) total += nums[i];
        double maxi = total;
        for (int i = k; i < nums.size(); ++i){
            total -= nums[i - k];
            total += nums[i];
            maxi = max(maxi, total);
        }
        return maxi / k;
    }
};